src/TBuffer.cpp

```diff
@@ -2253,6 +2253,64 @@ void TBuffer::append(const QString& text, int sub_start, int sub_end, TChar form
     }
 }
 
+// Wraps text to max line length of mWrapAt
+// Applies indentation of mWrapIndent to wrapped lines
+QString TBuffer::wrapText(const QString& text) const
+{
+    if (mWrapAt <= mWrapIndent) {
+        qWarning() << "mWrapAt (" << mWrapAt << ") is too small to accommodate mWrapIndent (" << mWrapIndent << ")";
+    }
+
+    const QString wordBreaks = qsl(",.- ");
+    QString wrappedText;
+    QString currentWord;
+    QString currentLine;
+    int wordsInCurrentLine = 0;
+
+    for (int i = 0; i < text.size(); i++) {
+        bool const at_newline = text.at(i) == '\n';
+        if (at_newline) {
+            currentLine += currentWord;
+            wrappedText += '\n' + currentLine;
```

Comment:
Suggestion, use the "QStringBuilder" https://wiki.qt.io/Using_QString_Effectively#QStringBuilder_:_Fast_QString_concatenation to reduce the number of `QString`s that need to be constructed:
```suggestion
            wrappedText += QChar::LineFeed % currentLine;
```